.github/workflows/publish-auxiliary-packages.yml

```diff
@@ -0,0 +1,54 @@
+name: Publish Auxiliary Packages
+on:
+  # NOTE: avoid using the manual execution unless is the only way to fix some issue,
+  # rather retry failed jobs in case of flakes. The manual execution can potentially
+  # cause the tags to point to a different commit that the one used to publish
+  # the packages
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  publish:
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+        startsWith(github.event.head_commit.message, 'chore(release): bump auxiliary package versions')
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Create Github App Token
+      uses: mongodb-js/devtools-shared/actions/setup-bot-token@main
+      id: app-token
+      with:
+        app-id: ${{ vars.DEVTOOLS_BOT_APP_ID }}
+        private-key: ${{ secrets.DEVTOOLS_BOT_PRIVATE_KEY }}
```

Comment:
Why do we need this? Is it because publish-auxiliary pushes some tags that we want to configure the git user for?

Reply:
I wasn't sure if the git workflow can just push anything at all. To be honest the relationship between the bot token and just an action is a bit unclear to me, might be nice to have a quick explainer on that sometime